snorkel/labeling/apply/core.py

```diff
@@ -11,6 +11,27 @@
 RowData = List[Tuple[int, int, int]]
 
 
+class ApplierMetadata(NamedTuple):
+    """Metadata about Applier call."""
+
+    faults: Dict[str, int]
+
+
+class _FunctionCaller:
+    def __init__(self, fault_tolerant: bool):
+        self.fault_tolerant = fault_tolerant
+        self.fault_counts: DefaultDict[str, int] = DefaultDict(int)
+
+    def __call__(self, f: LabelingFunction, x: DataPoint) -> int:
```

Comment:
do we want support for `f(*args, **kwargs)`, in case folks want to pass preprocessors/resources?

Reply:
can abstract later if needed, but let's leave for now

Reply:
this seems like a pretty general case that we'll need to address soon, agree that it can be a different PR